### PR TITLE
fix: address multiple tags for containerize

### LINKED ISSUE #2789: bug: bentoml containerize should support multiple tags
### Describe the bug

The method `buildx.build` support a list of tags [here](https://github.com/bentoml/BentoML/blob/6b80bb533887d219c5bc2d8c01ea1d8dfdb13dc7/bentoml/_internal/utils/buildx.py#L168) but `containerize` only send a single tag [here](https://github.com/bentoml/BentoML/blob/6b80bb533887d219c5bc2d8c01ea1d8dfdb13dc7/bentoml/bentos.py#L423).

This causes that when passing multiple `--docker-image-tag` arguments to `bentoml containerize`, only the last one is kept.

### To reproduce

Pass multiple `--docker-image-tag` to `bentoml containerize` and only the last one will be kept.

### Expected behavior

In my opinion, if multiple `--docker-image-tag` are passed to `bentoml containerize`, all of them should be passed to buildx.

### Environment

`bentoml`: 1.0.0
`python`: 3.9.10
`platform`: macOS-12.5-arm64-arm-64bit

### PR DESCRIPTION
**Pull request by aarnphm**
Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

## What does this PR address?

Fixes #2789

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
  those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

## Who can help review?

cc @parano

### CODE REVIEW

File: bentoml/_internal/cli/containerize.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ def add_containerize_command(cli: click.Group) -> None:
         help="Name and optionally a tag (format: 'name:tag'), defaults to bento tag.",
         required=False,
         callback=validate_tag,
+        multiple=True,
```
Comment (sauyon) [inline bentoml/_internal/cli/containerize.py:36]: Does this not change the type for the variable? Do we expect a list already?

**Review by bojiang — DISMISSED** [review]

File: bentoml/_internal/utils/docker.py [hunk 2]
```diff
@@ -67,3 +63,16 @@ def validate_tag(
...
+
+def validate_tag(
+    ctx: "click.Context", param: "click.Parameter", tag: str | tuple[str] | None
+) -> str | None:
+    if tag is None:
+        return tag
+    elif isinstance(tag, tuple):
+        for t in tag:
+            assert not isinstance(t, list)
+            return validate_tag(ctx, param, t)
```
Comment (bojiang) [inline bentoml/_internal/utils/docker.py:76]:
```suggestion
        return all(validate_tag(t) for t in tag)
```

File: bentoml/bentos.py [hunk 3]
```diff
@@ -420,7 +420,7 @@ def containerize(
     buildx.health()
 
     bento = _bento_store.get(tag)
-    if docker_image_tag is None:
+    if not docker_image_tag:
         docker_image_tag = str(bento.tag)
```
Comment (ssheng) [inline bentoml/bentos.py:424]: What if `docker_image_tag` is a list here?

Comment (aarnphm) [reply bentoml/bentos.py:424]: if `docker_image_tag` is a list, then we don't generate a image using bento.tag

**Review by ssheng — APPROVED** [review]

**Pull request merged**